test/junit/scala/collection/immutable/ListTest.scala

```diff
@@ -75,4 +75,18 @@ class ListTest {
   }
 
   @Test def checkSearch: Unit = SeqTests.checkSearch(List(0 to 1000: _*), 15,  implicitly[Ordering[Int]])
+
+
+  @Test def colonColonColon(): Unit = {
+
+    assertEquals(Nil, Nil ::: Nil)
```

Comment:
it may be worth adding some `assertNonAllocating` on the appropriate cases

Reply:
I'm not familiar with this method, what's the import? Or are you saying I could implement it?

Reply:
I misremembered the name - its just `nonAllocating`

see `HashSetTest`
your test class needs to extend `scala.tools.testing.AllocationTest`
then you can call 
```
  def nonAllocating[T: Manifest](fn: => T)(implicit execution: AllocationExecution = AllocationExecution()): T 

  def onlyAllocates[T: Manifest](size: Int)(fn: => T)(implicit execution: AllocationExecution = AllocationExecution()): T 
``` 

so you can get tests like this (in HashSetTest)

```

  @Test
  def nonAllocatingEqualsNotShared(): Unit = {
    val base = generate()
    val notShared = generate()

    assertTrue(nonAllocating {
      base == notShared
    })
    assertTrue(nonAllocating {
      notShared == base
    })
  }
```